src/rgw/rgw_rest_s3.h

```diff
@@ -873,6 +874,72 @@ static inline int valid_s3_bucket_name(const string& name, bool relaxed=false)
   return 0;
 }
 
+namespace s3selectEngine
+{
+class s3select;
+class csv_object;
+class RGWSelectObj_ObjStore_S3 : public RGWGetObj_ObjStore_S3
```

Comment:
the `s3selectEngine` namespace is provided by your `s3select` library. its fine to forward declare its types here, but radosgw shouldn't define new types in it. so i would move `class RGWSelectObj_ObjStore_S3` outside the namespace